quarkus/admin/src/main/java/org/apache/polaris/admintool/config/QuarkusJdbcDataSource.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.admintool.config;
+
+import io.quarkus.arc.InstanceHandle;
+import java.util.List;
+import javax.sql.DataSource;
+import org.apache.polaris.extension.persistence.relational.jdbc.JdbcDatasource;
+import org.postgresql.shaded.com.ongres.scram.common.util.Preconditions;
+
+public class QuarkusJdbcDataSource implements JdbcDatasource {
+  private final List<InstanceHandle<DataSource>> dataSources;
+  private final String realmIsolationType;
+
+  public static final String DEFAULT_DATA_SOURCE_NAME = "<default>";
+  public static final String REALM_ISOLATION_TYPE_DB = "db";
+
+  public QuarkusJdbcDataSource(String realmIsolationType, List<InstanceHandle<DataSource>> dataSources) {
+    this.realmIsolationType = realmIsolationType;
+    this.dataSources = dataSources;
+  }
+
+  @Override
+  public DataSource fromRealmId(String realmId) {
+    Preconditions.checkNotNull(REALM_ISOLATION_TYPE_DB.equals(realmIsolationType) && dataSources.size() == 1, "Misconfigured datasources as per realm requirements");
+    for (InstanceHandle<DataSource> handle : dataSources) {
+      String name = handle.getBean().getName();
+      name = name == null ? DEFAULT_DATA_SOURCE_NAME : unquoteDataSourceName(name);
+      // if realm isolation is DB then there should be only one DS configured.
+      if (name.equals(realmId) || (REALM_ISOLATION_TYPE_DB.equals(realmIsolationType) && name.equals(DEFAULT_DATA_SOURCE_NAME))) {
+        return handle.get();
+      }
+    }
+    throw new IllegalStateException("No datasource configured with name: " + realmId);
+  }
+
+  public static String unquoteDataSourceName(String dataSourceName) {
+    if (dataSourceName.startsWith("\"") && dataSourceName.endsWith("\"")) {
```

Comment:
It might be best to define a config `Map` for translating realm ID to DS names (cf. `RealmContextConfiguration`), defaulting to 1:1 association when not configured.